### PR TITLE
Revise search backpressure documentation for 2.6

### LINKED ISSUE #2629: [DOC] Search backpressure cluster settings documentation
**What do you want to do?**
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 As a part of [Search backpressure cancellation at the coordinator level](https://github.com/opensearch-project/OpenSearch/issues/5173), we have introduced a new dynamic cluster settings and deprecated a few existing dynamic settings. Using the new settings, customers can tune the search backpressure thresholds.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
Issue link: https://github.com/opensearch-project/OpenSearch/issues/5173
PR: https://github.com/opensearch-project/OpenSearch/pull/5605
Current public doc: https://opensearch.org/docs/latest/opensearch/search-backpressure#search-backpressure-settings

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #2629

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by PritLadani — APPROVED** [review]
LGTM.

File: _tuning-your-cluster/availability-and-recovery/search-backpressure.md [hunk 1]
```diff
@@ -81,19 +81,30 @@ Search backpressure adds several settings to the standard OpenSearch cluster set
...
+search_backpressure.search_task.cancellation_burst | 5 | The maximum number of search tasks to cancel in a single iteration of the observer thread.
+search_backpressure.search_task.heap_percent_threshold | 2% | The heap usage threshold (as a percentage) required for an individual parent task before it is considered for cancellation.
+search_backpressure.search_task.total_heap_percent_threshold | 5% | The heap usage threshold (as a percentage) required for the sum of heap usages of all search tasks before cancellation is applied.
+search_backpressure.search_task.heap_variance | 2.0 | The heap usage variance required for an individual parent task before it is considered for cancellation. A task is considered for cancellation when `taskHeapUsage` is greater than or equal to `heapUsageMovingAverage` * `variance`.
+search_backpressure.search_task.heap_moving_average_window_size | 10 | The window size to calculate the rolling average of the heap usage for the completed parent tasks.
+search_backpressure.search_task.cpu_time_millis_threshold | 30,000 | The CPU usage threshold (in milliseconds) required for an individual parent task before it is considered for cancellation.
```
Comment (carolxob) [inline _tuning-your-cluster/availability-and-recovery/search-backpressure.md:98]: Global suggestion: remove the parentheses and add commas.

```suggestion
search_backpressure.search_task.cpu_time_millis_threshold | 30,000 | The CPU usage threshold, in milliseconds, required for an individual parent task before it is considered for cancellation.
```

**Review by carolxob — APPROVED** [review]
LGTM other than one minor suggestion.

**Review by ariamarble — APPROVED** [review]
Looks good other than single comment.

File: _tuning-your-cluster/availability-and-recovery/search-backpressure.md (hunk 1 above)
Comment (kolchfa-aws) [reply _tuning-your-cluster/availability-and-recovery/search-backpressure.md:98]: This is the old wording from the original PR that has been through editorial. Let's ask @natebower's opinion.

File: _tuning-your-cluster/availability-and-recovery/search-backpressure.md [hunk 2]
```diff
@@ -81,19 +81,30 @@ Search backpressure adds several settings to the standard OpenSearch cluster set
...
+search_backpressure.search_task.cancellation_rate| 0.003 | The maximum number of search tasks to cancel per millisecond of elapsed time.
+search_backpressure.search_task.cancellation_burst | 5 | The maximum number of search tasks to cancel in a single iteration of the observer thread.
+search_backpressure.search_task.heap_percent_threshold | 2% | The heap usage threshold (as a percentage) required for an individual parent task before it is considered for cancellation.
+search_backpressure.search_task.total_heap_percent_threshold | 5% | The heap usage threshold (as a percentage) required for the sum of heap usages of all search tasks before cancellation is applied.
+search_backpressure.search_task.heap_variance | 2.0 | The heap usage variance required for an individual parent task before it is considered for cancellation. A task is considered for cancellation when `taskHeapUsage` is greater than or equal to `heapUsageMovingAverage` * `variance`.
+search_backpressure.search_task.heap_moving_average_window_size | 10 | The window size to calculate the rolling average of the heap usage for the completed parent tasks.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/search-backpressure.md:97]:
```suggestion
search_backpressure.search_task.heap_moving_average_window_size | 10 | The window size used to calculate the rolling average of the heap usage for the completed parent tasks.
```

File: _tuning-your-cluster/availability-and-recovery/search-backpressure.md (hunk 1 above)
Comment (natebower) [reply _tuning-your-cluster/availability-and-recovery/search-backpressure.md:98]: They are both grammatically correct and essentially serve the same purpose, so this is a matter of subjective preference.

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Just one added word :-)

**Pull request merged**